src/views/MVI/components/MarketData.tsx

```diff
@@ -0,0 +1,132 @@
+import React, { useState, useEffect } from 'react'
```

Comment:
Besides certain things like image assets, shouldn't this be pretty similar across different products? The reason I ask is because I think there may be an opportunity to abstract most of the functionality encapsulated by the component.

Reply:
Yup, definitely. I think we should be able to pass in price data, token image url to a single interface and get most of this stuff out